chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java

```diff
@@ -837,7 +848,39 @@ public synchronized void loadChunks(TaskTracker taskTracker, World world, Collec
 
         loadedChunks.add(cp);
 
-        world.getChunk(cp).getChunkData(chunkData, palette);
+        try {
+          nextChunkDataTask.get();
+        } catch(InterruptedException ignored) { // If except, load the chunk synchronously
+          if(usingFirstChunkData) {
+            world.getChunk(chunkPositions[i]).getChunkData(chunkData1, palette);
```

Comment:
Is the `BlockPalette` thread-safe, i.e. are we allowed to get blocks from the hash maps while other threads add new blocks to it?

Reply:
definitely not, I'll fix this today